src/game_battler.cpp

```diff
@@ -437,7 +437,12 @@ int Game_Battler::ApplyConditions() {
 }
 
 void Game_Battler::RemoveBattleStates() {
-	State::RemoveAllBattle(GetStates(), GetPermanentStates());
+	std::vector<bool> removed_states = State::RemoveAllBattle(GetStates(), GetPermanentStates());
+
+	if (!removed_states.empty() && removed_states[0]) {
+		// Death was removed
+		SetHp(1);
+	}
```

Comment:
Instead of returning `vector<bool>` and allocating memory, you could just do:

1. was_dead = isDead()
2. RemoveAllBattle()
3. if (was_dead && !isDead()) { ... }